test/hitchhiker/konserve_test.clj

```diff
@@ -0,0 +1,131 @@
+                             (fn [{:keys [children cfg op-buf]}]
+                               (core/->IndexNode (->> children
+                                                      vec)
+                                                 (promise-chan)
+                                                 (vec op-buf)
+                                                 cfg))
+                             'hitchhiker.tree.messaging.InsertOp
+                             msg/map->InsertOp
+                             'hitchhiker.tree.messaging.DeleteOp
+                             msg/map->DeleteOp
+                             'hitchhiker.tree.core.Config
+                             core/map->Config})
+      store))
+
+
+(deftest simple-konserve-test
+  (testing "Insert and lookup"
+    (let [folder "/tmp/async-hitchhiker-tree-test"
+          store (setup-store folder)
+          backend (kons/->KonserveBackend store)
+          flushed (<?? (core/flush-tree
+                        (time (reduce (fn [t i]
+                                        (<?? (msg/insert t i i)))
+                                      (<?? (core/b-tree (core/->Config 1 3 (- 3 1))))
+                                      (range 1 11)))
+                        backend))
+          root-key (kons/get-root-key (:tree flushed))
+          tree (<?? (kons/create-tree-from-root-key store root-key))]
+      (is (= (<?? (msg/lookup tree -10)) nil))
+      (is (= (<?? (msg/lookup tree 100)) nil))
+      (dotimes [i 10]
+        (is (= (<?? (msg/lookup tree (inc i))) (inc i))))
+      (is (= (map first (msg/lookup-fwd-iter tree 4)) (range 4 11)))
+      (is (= (map first (msg/lookup-fwd-iter tree 0)) (range 1 11)))
+      (let [deleted (<?? (core/flush-tree (<?? (msg/delete tree 3)) backend))
+            root-key (kons/get-root-key (:tree deleted))
+            tree (<?? (kons/create-tree-from-root-key store root-key))]
+        (is (= (<?? (msg/lookup tree 2)) 2))
+        (is (= (<?? (msg/lookup tree 3)) nil))
+        (is (= (<?? (msg/lookup tree 4)) 4)))
+      (delete-store folder))))
+
+
+
+;; adapted from redis tests
+
+(defn insert
+  [t k]
+  (msg/insert t k k))
+
+(defn lookup-fwd-iter
+  [t v]
+  (seq (map first (msg/lookup-fwd-iter t v))))
+
+(defn mixed-op-seq
+  "This is like the basic mixed-op-seq tests, but it also mixes in flushes to a konserve filestore"
+  [add-freq del-freq flush-freq universe-size num-ops]
+  (prop/for-all [ops (gen/vector (gen/frequency
+                                   [[add-freq (gen/tuple (gen/return :add)
+                                                         (gen/no-shrink gen/int))]
+                                    [flush-freq (gen/return [:flush])]
+                                    [del-freq (gen/tuple (gen/return :del)
+                                                         (gen/no-shrink gen/int))]])
+                                 num-ops)]
+                #_(assert (let [ks (wcar {} (car/keys "*"))]
```

Comment:
I wonder if this assert could still be ported--the idea is to check that the backend is starting empty.